algorithm.go

```diff
@@ -164,3 +173,23 @@ func (a Algorithm) FromBytes(p []byte) Digest {
 func (a Algorithm) FromString(s string) Digest {
 	return a.FromBytes([]byte(s))
 }
+
+// Validate validates the encoded portion string
+func (a Algorithm) Validate(encoded string) error {
+	r, ok := anchoredEncodedRegexps[a]
+	if !ok {
+		return ErrDigestUnsupported
```

Comment:
> This validates the character set for the encoded portion for the algorithm. Nothing less, nothing more. If you use a function incorrectly, you pay the price.

But if your `Algorithm` is not listed in `anchoredEncodedRegexps`, we can still validate the encoded portion of the algorithm using the [generic spec charset][1] and distinguish between “I don't know about your algorithm; maybe the encoded part here is ok” and “I don't know about your algorithm but I know your encoded part is wrong”.

[1]: https://github.com/opencontainers/image-spec/blob/f03dbe35d449c54915d235f1a3cf8f585a24babe/descriptor.md#digests